### PR TITLE
[14.0][FIX] purchase_invoice_plan: error when invoice=False

### PR DESCRIPTION
**Pull request by ps-tubtim**
- Fix error when `invoice = False`
- Add hook function for computing `amount_invoice`

### CODE REVIEW

File: purchase_invoice_plan/models/purchase.py [hunk 1]
```diff
@@ -271,7 +271,11 @@ def _compute_invoiced(self):
                 lambda l: l.state in ("draft", "posted")
             )
             rec.invoiced = invoiced and True or False
-            rec.amount_invoiced = invoiced[:1].amount_untaxed
+            positive_lines = invoiced[:1].invoice_line_ids.filtered(
+                lambda l: l.price_subtotal > 0
```
Comment (kittiu) [inline purchase_invoice_plan/models/purchase.py]: I think we should check with is_deposit flag (which can relate through po_line, with module purchase_invoice_plan_deposit).

Checking with negative amount, we can't be sure as user may just want to give discount, etc...

BUT, we may need to make a Hook functon to get the amount_invoiced.

**Review by kittiu — APPROVED** [review]
LGTM

**Review by Saran440 — APPROVED** [review]

**Review by HviorForgeFlow — APPROVED** [review]
Code review

**Pull request merged**